api/app/Models/PoolCandidate.php

```diff
@@ -938,14 +938,28 @@ public function computeAssessmentStatus()
             $hasToAssess = false;
 
             $stepResults = $this->assessmentResults->where('assessment_step_id', $stepId);
-            if ($stepResults->isEmpty()) {
-                continue;
-            }
-
             $hasEssentialSkillsToAssess = $step->poolSkills->contains(function ($poolSkill) {
                 return $poolSkill->type === PoolSkillType::ESSENTIAL->name;
             });
 
+            // If no essential skills are to be assessed
+            // and not on application screening (requires education assessment)
+            // then it is an automatic pass
+            if (! $hasEssentialSkillsToAssess && $step->type !== AssessmentStepType::APPLICATION_SCREENING->name) {
+                $decisions[] = [
+                    'step' => $stepId,
+                    'decision' => AssessmentDecision::SUCCESSFUL->name,
+                ];
+
+                $currentStep++;
+
+                continue;
+            }
+
+            if ($stepResults->isEmpty()) {
+                continue;
```

Comment:
I don't think we can continue without adding _something_ to the decisions array. If we ended up here, that means there _are_ skill or education requirements, but no results, meaning we should add a TO_ASSESS decision, right?

Reply:
Yeah, that makes sense. I think it was also required for the other comment you made so i rolled it into [do not increment step when previous steps not passed](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10642/commits/943049c819e067eddbd8a9a6a07310c5ca030780)